Update Serial API for new-style event pages

Elchi3 opened the pull request:
This PR updates the Serial API for new-style event pages. (see the [template](https://developer.mozilla.org/en-US/docs/MDN/Structures/Page_types/API_event_subpage_template))

Commands used:

```
yarn content move Web/API/SerialPort/onconnect Web/API/SerialPort/connect_event
yarn content move Web/API/SerialPort/ondisconnect Web/API/SerialPort/disconnect_event
yarn content delete Web/API/Serial/onconnect --redirect=/en-US/docs/Web/API/SerialPort/connect_event 
yarn content delete Web/API/Serial/ondisconnect --redirect=/en-US/docs/Web/API/SerialPort/disconnect_event
```

As discussed in https://github.com/mdn/browser-compat-data/pull/13925, `SerialPort` bubbles to `Serial` and so I documented these two events only under `SerialPort` and the `Serial` page just contains them as links but not as sub pages to avoid duplication.

Elchi3 committed:
Update Serial API for new-style event pages

wbamberg commented on files/en-us/web/api/serialport/connect_event/index.md:
```diff
@@ -0,0 +1,51 @@
+
+A generic {{domxref("Event")}}. 
+
+Bubbles to {{domxref("Serial")}} (`event.port` is available representing the {{domxref("SerialPort")}}).
```
As we said, it's weird that this is an `Event` but has a `port` property.

teoli2003 replied:
This is needed if it bubbles up to `Navigator`or `Serial`, so that the event handler knows which port it relates to.

Elchi3 replied:
It is weird and I think I got this wrong.  In fact, I think the spec is wrong. I filed https://github.com/WICG/serial/pull/159

See also https://web.dev/serial/#connection-disconnection which says that this used to be a `SerialConnectionEvent` with a `port` property. That's no longer the case though.

wbamberg commented on files/en-us/web/api/serialport/connect_event/index.md:
````diff
@@ -0,0 +1,51 @@
+## Examples
+
+```js
+navigator.serial.addEventListener('connect', event => {
````
This looks like an example for `Serial` not `SerialPort`.

Elchi3 replied:
I've now made two examples to show both.

wbamberg commented on files/en-us/web/api/serialport/disconnect_event/index.md:
```diff
@@ -0,0 +1,51 @@
+
+A generic {{domxref("Event")}}. 
+
+Bubbles to {{domxref("Serial")}} (`event.port` is available representing the {{domxref("SerialPort")}}).
```
Same comment as for `connect`.

wbamberg commented on files/en-us/web/api/serialport/disconnect_event/index.md:
````diff
@@ -0,0 +1,51 @@
+## Examples
+
+```js
+navigator.serial.addEventListener('disconnect', event => {
````
Again, same comment as for `connect`.

wbamberg reviewed (changes requested):
This looks fine, except that the examples are AFAICT for `Serial`, not `SerialPort`, and as we discussed, the `event.port` thing is odd.

teoli2003 commented:
Thinking out loud about bubbling.

If I understand well, the hierarchy here is: (- means "contains", not "inherits")

(`Worker`)`Navigator` - `Serial` - `SerialPort`.

So an event will be thrown to `SerialPort`, then bubble to `Serial`, and then to (Worker)Navigator. I see they are listed in Serial, but not on (Worker)Navigator.

I think this may be a too large list if we do the same in Navigator. But maybe we could have in (Worker)Navigator:

=== Events ===
_Events bubbling from `Serial` may be dispatched to this object._ (We don't need the detailed list there)

And in Serial:

_If not caught, these events will bubble up the tree (to `Navigator` or `WorkerNavigator`)._

(Bubbling in the DOM Tree would have: _If not caught, these events will bubble up the DOM Tree_)

I think we need to know if an event will bubble if not caught on a specific object.

teoli2003 commented:
I would add that it is not trivial that non-caught bubbling events on `Serial` will bubble to `Navigator` (or `WorkerNavigator`)

Elchi3 committed:
Clarify bubbling; update examples

Elchi3 commented:
Thanks @wbamberg and @teoli2003, ready for re-review. Corrected the bubbling section and added examples for showing both, bubbling to `Serial` and non-bubbling with `SerialPort`.

wbamberg reviewed (approved):
👍 lgtm, thanks Florian!

Elchi3 committed:
Merge branch 'main' into events-serial

Elchi3 committed:
move inheritance diagram to the right spot
